esrally/track/params.py

```diff
@@ -43,18 +43,24 @@ def param_source_for_operation(op_type, track, params, task_name):
 def param_source_for_name(name, track, params):
     param_source = __PARAM_SOURCES_BY_NAME[name]
 
-    # we'd rather use callable() but this will erroneously also classify a class as callable...
-    if isinstance(param_source, types.FunctionType):
+    if inspect.isfunction(param_source):
         return DelegatingParamSource(track, params, param_source)
     else:
         return param_source(track, params)
 
 
+def ensure_valid_param_source(param_source):
+    if not inspect.isfunction(param_source) and not inspect.isclass(param_source):
+        raise exceptions.RallyAssertionError(f"Parameter source [{param_source}] must be either a function or a class.")
+
```

Comment:
nit: 2 spaces or one between methods? We seem to have an inconsistency

Reply:
Thanks for pointing this out.

Unfortunately, the linter that we have configured for this project (`pylint`) cannot check this but I ran [pycodestyle](https://pypi.org/project/pycodestyle/) against the code base locally and only found one spot where we only use one line instead of two lines:

```
(.venv) daniel@io:Projects/rally ‹param-source-check›$ pycodestyle esrally | grep E302
esrally/track/loader.py:726:1: E302 expected 2 blank lines, found 1
```

I'll fix that separately.